gravitee-apim-gateway/gravitee-apim-gateway-http/src/main/java/io/gravitee/gateway/jupiter/http/vertx/VertxHttpServerRequest.java

```diff
@@ -36,20 +30,20 @@
  */
 public class VertxHttpServerRequest extends AbstractVertxServerRequest {
 
-    private final BodyChunksFlowable bodyChunksFlowable;
+    private final BufferFlow bufferFlow;
+    private MessageFlow messageFlow;
     protected Boolean isWebSocket = null;
     protected WebSocket webSocket;
-    private final MessageFlowable httpMessages;
 
     public VertxHttpServerRequest(final HttpServerRequest nativeRequest, IdGenerator idGenerator) {
         super(nativeRequest, idGenerator);
-        bodyChunksFlowable = new BodyChunksFlowable();
-        bodyChunksFlowable.chunks =
+        bufferFlow = new BufferFlow();
+        bufferFlow.chunks =
             nativeRequest
                 .toFlowable()
                 .doOnNext(buffer -> metrics().setRequestContentLength(metrics().getRequestContentLength() + buffer.length()))
                 .map(Buffer::buffer);
-        httpMessages = new MessageFlowable();
+        messageFlow = null;
```

Comment:
Why do you always init `bodyChunksFlowable` while you are not for `MessageFlowable`? Would it be easier to alsways init both?

Reply:
No because initializing message flow allows to know that we switch from chunk to message. Makes senses ?